shm/game.cpp

```diff
@@ -0,0 +1,186 @@
+#include "game.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+
+#include "cargo.hpp"
+#include "island.hpp"
+#include "store.hpp"
+
+Game::Game(size_t money, size_t days, size_t finalGoal)
+    : money_(money),
+      days_(days),
+      finalGoal_(finalGoal),
+      currentDay_(1),
+      publisher_(std::make_shared<Time>()),
+      ship_(std::make_shared<Ship>(150, 50, 13, "Player_ship", 1, publisher_)),
+      player_(std::make_shared<Player>(ship_, 500, 200, publisher_)),
+      time_(std::make_shared<Time>()),
+      map_(std::make_shared<Map>(time_.get())) {}
+
+void Game::startGame() {
+    printTrail();
+    printStars();
+    printStars();
+    std::cout << std::setw(8) << "Welcome" 
+              << std::setw(8) << "in" 
+              << std::setw(8) << "SHM" 
+              << std::setw(8) << "Game"
+              << std::setw(8) << "v1.0"
+              << std::setw(8);
+    printStars();
+    printStars();
+    std::cout << '\n';
+    printTrail();
+    printStars();
+    std::cout << std::setw(8) << "You have: " << days_ << " days to get: " << finalGoal_ << " gold. Have a nice time!";
+    printStars();
+    printStars();
+    std::cout << "\n\n\n";
+    std::cout << "~~~May the force be with you!~~~ \n";
+    printTrail();
```

Comment:
You can create separate function like: Print welcome page or sth like this. This function is definitely too long.

Reply:
right, like first programs in int main() when you don't know how to create different functions than main :D